src/openms_gui/source/VISUAL/LayerData.cpp

```diff
@@ -219,7 +220,8 @@ namespace OpenMS
     // no ID selected
     if (peptide_id_index == -1 || peptide_hit_index == -1) { return; }
 
-    MSSpectrum & spectrum = getCurrentSpectrum();
+    // TODO: check if this is correct?
+    MSSpectrum & spectrum = getCurrentSpectrumMuteable();
```

Comment:
@timosachsenberg can you check whether this should be `MSSpectrum & spectrum = getPeakDataMuteable()->getSpectrum(current_spectrum_);`